moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_arm.h

```diff
@@ -78,20 +93,46 @@ class JogInterfaceBase
   bool changeControlDimensions(moveit_msgs::ChangeControlDimensions::Request& req,
                                moveit_msgs::ChangeControlDimensions::Response& res);
 
-  // Jogging calculation thread
-  bool startJogCalcThread();
+  /** \brief Provide a Cartesian velocity command to the jogger.
+   * The units are determined by settings in the yaml file.
+   */
+  void provideTwistStampedCommand(const geometry_msgs::TwistStamped& velocity_command);
+  void provideTwistStampedCommand(const geometry_msgs::TwistStampedConstPtr& msg);
 
-  /** \brief Stop the main calculation thread */
-  bool stopJogCalcThread();
+  /** \brief Send joint position(s) commands */
+  void provideJointCommand(const control_msgs::JointJog& joint_command);
+  void provideJointCommand(const control_msgs::JointJogConstPtr& msg);
```

Comment:
Please fill me in on why you overloaded this

Reply:
I did this so I can use the same method name for the callback for a ros subscriber and the C++ interface.  If I convert this to use ros style inter-process communication I'd probably change it to only have one method (the Ptr one) and call it a callback and make people using the C++ interface publish to the topic.

Reply:
`ros::Subscriber` is perfectly capable of handling a callback with a single `const pkg_msgs::Msg&` parameter.
This is usually the better option if you do not intend to store the `shared_ptr` somewhere.

Reply:
See the latest version of this where I only accept the shared ptr for these as I do store the lastest one I've received.

Reply:
:laughing: I did not expect a reply for the next 6 hours at least. ;)

You are right and it's the correct interface now. I've seen a lot of code requiring the `ConstPtr&` for no particular reason over the years, so I thought I'd mention the `const Msg&` interface.